src/EventStore.Core/Services/PersistentSubscription/PersistentSubscriptionStreamReader.cs

```diff
@@ -46,35 +46,51 @@ public class PersistentSubscriptionStreamReader : IPersistentSubscriptionStreamR
 				_ioDispatcher.ReadForward(
 					eventSource.EventStreamId, startPosition.StreamEventNumber, Math.Min(countToLoad, actualBatchSize),
 					resolveLinkTos, SystemAccounts.System, new ResponseHandler(onEventsFound, onError, skipFirstEvent).FetchCompleted,
-					async () => {
-						var backOff = GetBackOffDelay(retryCount);
-						Log.Warning("Timed out reading from stream: {stream}. Retrying in {retryInterval} seconds.", eventSource.EventStreamId, backOff);
-						await Task.Delay(TimeSpan.FromSeconds(backOff)).ConfigureAwait(false);
-						BeginReadEventsInternal(eventSource, startPosition, countToLoad, batchSize, resolveLinkTos,
-						skipFirstEvent, onEventsFound, onError, retryCount + 1);
-					}, Guid.NewGuid());
-
+					async () => await HandleTimeout(eventSource.EventStreamId).ConfigureAwait(false),
+					Guid.NewGuid());
 			} else if (eventSource.FromAll) {
-				_ioDispatcher.ReadAllForward(
-					startPosition.TFPosition.Commit,
-					startPosition.TFPosition.Prepare,
-					Math.Min(countToLoad, actualBatchSize),
-					resolveLinkTos,
-					true,
-					null,
-					SystemAccounts.System,
-					null,
-					new ResponseHandler(onEventsFound, onError, skipFirstEvent).FetchAllCompleted,
-					async () => {
-						var backOff = GetBackOffDelay(retryCount);
-						Log.Warning("Timed out reading from stream: {stream}. Retrying in {retryInterval} seconds.", SystemStreams.AllStream, backOff);
-						await Task.Delay(TimeSpan.FromSeconds(backOff)).ConfigureAwait(false);
-						BeginReadEventsInternal(eventSource, startPosition, countToLoad, batchSize, resolveLinkTos,
-							skipFirstEvent, onEventsFound, onError, retryCount + 1);
-					},Guid.NewGuid());
+				if (eventSource.EventFilter is null) {
+					_ioDispatcher.ReadAllForward(
+						startPosition.TFPosition.Commit,
+						startPosition.TFPosition.Prepare,
+						Math.Min(countToLoad, actualBatchSize),
+						resolveLinkTos,
+						true,
+						null,
+						SystemAccounts.System,
+						null,
+						new ResponseHandler(onEventsFound, onError, skipFirstEvent).FetchAllCompleted,
+						async () => await HandleTimeout(SystemStreams.AllStream).ConfigureAwait(false),
+						Guid.NewGuid());
+				} else {
+					_ioDispatcher.ReadAllForwardFiltered(
+						startPosition.TFPosition.Commit,
+						startPosition.TFPosition.Prepare,
+						Math.Min(countToLoad, actualBatchSize),
+						resolveLinkTos,
+						true,
+						actualBatchSize, // TODO: We need to set the max window size for the subscriptions
```

Comment:
Did you forget to do it or it's a reminder for another time?

Reply:
It was something I forgot to do, thanks!